app/Infrastructure/Controllers/PostOpenWalletController.php

```diff
@@ -2,40 +2,38 @@
 
 namespace App\Infrastructure\Controllers;
 
-use App\Application\DataSources\UserDataSource;
-use App\Application\DataSources\WalletDataSource;
+use App\Application\Services\OpenWalletService;
 use App\Http\Requests\OpenWalletRequest;
-use Illuminate\Http\Request;
 use Illuminate\Http\Response;
 use Illuminate\Http\JsonResponse;
 use Illuminate\Routing\Controller as BaseController;
-use Illuminate\Support\Facades\Validator;
 
 class PostOpenWalletController extends BaseController
 {
-    private UserDataSource $userDataSource;
-    private WalletDataSource $walletDataSource;
+    private OpenWalletService $openWalletService;
 
-    public function __construct(UserDataSource $userDataSource, WalletDataSource $walletDataSource)
+    public function __construct(OpenWalletService $openWalletService)
     {
-        $this->userDataSource = $userDataSource;
-        $this->walletDataSource = $walletDataSource;
+        $this->openWalletService = $openWalletService;
     }
     public function __invoke(OpenWalletRequest $request): JsonResponse
     {
-        $user = $this->userDataSource->findById($request->input('user_id'));
-        if (is_null($user)) {
+        $userId = $request->input('user_id');
+        $walletId = $this->openWalletService->openWallet($userId);
```

Comment:
Maybe we are duplicating the names? The service its called `openWallet...` and the method is openWallet